electron/main/Generic/path.ts

```diff
@@ -0,0 +1,25 @@
+import path from "path";
+
+export function addExtensionToFilenameIfNoExtensionPresent(
+  filename: string,
+  extension: string
+): string {
+  const trimmedFilename = filename.trim();
+  const trimmedExtension = extension.trim();
+
+  if (!trimmedFilename || !trimmedExtension) {
+    return trimmedFilename;
+  }
```

Comment:
Nit: if !trimmedFilename, perhaps we should throw an error

Reply:
Done